keps/2724-topology-aware-scheduling/README.md

```diff
@@ -977,6 +1053,48 @@ Explanation:
 
 It is worth noting that the tight fit mentioned above does not guarantee that no free capacity will be left within the assigned domains.
 
+### Cross-PodSet Topology Aware scheduling
+
+In consideration of a [Story 6](#story-6) a cross-podset topology aware scheduling
+is required, due to the fact that leader and workers are separate PodSets. To be able 
+to co-locate leaders with its workers in the same topology domain, we divide the
+problem into subproblems:
+
+- Ensure leader and workers end up on the same flavor
+- Find topology domain to co-locate leader and workers
+
+#### Ensure leader and workers end up on the same flavor
+
+To ensure leader and workers are assigned the same flavor a notion of PodSetGroup is introduced:
+
+```golang
+type PodSetTopologyRequest struct {
+  // ...
+
+	// PodSetGroup indicates the name of the group of PodSets to which this PodSet belongs to.
+	// PodSets with the same `PodSetGroup` should be assigned the same ResourceFlavor
+	//
+	// +optional
+	PodSetGroup *string `json:"podSetGroup,omitempty"`
+  
+  // ...
+}
+```
```

Comment:
```suggestion
To ensure leader and workers are assigned the same flavor a notion of PodSetGroup is 
introduced under the PodSetTopologyRequest's PodSetGroupName field (see [Internal APIs](Internal APIs)):
```
To avoid duplication